### DISCUSSION TITLE
Switch to Ruff?

### DISCUSSION THREAD

**Discussion by jduerholt** [meta-pytorch/botorch#2100] [Ideas]
I was just wondering what do you think about switching from different tools for linting and formatting (flake8, black, ufmt) to ruff (https://docs.astral.sh/ruff), as it provides the same functionality as the above mentioned tools in just one tool with faster execution time?

**Comment by Balandat** [comment]
That's an interesting suggestion, I wasn't aware of ruff. My main concern is that we also use the (flake8, black, ufmt) linting/formatting/sorting toolchain in our internal repos (which are the source of truth of the botorch code) and so it'll be much easier to maintain consistency between internal and external CI. So I don't expect this to happen anytime soon.

**Reply by jduerholt** [reply]
It happend: https://github.com/pytorch/botorch/pull/2582 ;) I like it!

**Reply by jduerholt** [reply]
I just sort that you also switch for usort to RUFF, should I provide a PR for it? Or do you want to do it internally?

**Reply by Balandat** [reply]
We're planning to also switch the sorting, but need to figure out a couple of things with internal systems first to make sure everything is in sync. Shouldn't be too long until we can make that change.

**Reply by jduerholt** [reply]
Sounds good! If you are starting to incorporate tools from Astral, it could also make sense to change the test pipelines to use `uv` (https://docs.astral.sh/uv/). I made very positive experience with it in using it in github test pipelines, as the installation of the packages is substantially faster than with plain pip. I could also provide a PR for it ;)
